### PR TITLE
Added async tests for blob

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/android/storage/blob/BlobTest.java [hunk 1]
```diff
@@ -288,6 +290,32 @@ public void getPropertiesError() {
...
+    public void getPropertiesAsync() {
+        // Setup
+        CountDownLatch latch = new CountDownLatch(1);
+
+        // When
+        asyncClient.getBlobProperties(containerName, blobName, new CallbackWithHeader<Void, BlobGetPropertiesHeaders>() {
+                @Override
+                public void onSuccess(Void result, BlobGetPropertiesHeaders header, Response response) {
+                    assertEquals(200, response.code());
+                    latch.countDown();
```
Comment (vcolin7) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/android/storage/blob/BlobTest.java:303]: Should we check that the header's contents are correct like we check the body contents in `rawRownloadAsync()`?

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/android/storage/blob/BlobTest.java [hunk 2]
```diff
@@ -929,4 +1042,30 @@ public void onFailure(Throwable throwable, Response response) {
...
+    @Test
+    public void getTagsAsync() {
+        // Setup
+        CountDownLatch latch = new CountDownLatch(1);
+
+        // When
+        asyncClient.getBlobTags(containerName, blobName, new CallbackWithHeader<Map<String, String>, BlobGetTagsHeaders>() {
+                @Override
+                public void onSuccess(Map<String, String> result, BlobGetTagsHeaders header, Response response) {
+                    assertEquals(200, response.code());
```
Comment (vcolin7) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/android/storage/blob/BlobTest.java:1055]: Should we check that the result's contents are correct like we check the body contents in `rawRownloadAsync()`?

**Review by vcolin7 — COMMENTED** [review]
In general it LGTM, just had a couple questions :)

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/android/storage/blob/BlobTest.java (hunk 1 above)
Comment (rickle-msft) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/android/storage/blob/BlobTest.java:303]: I think we have sync tests that validate the header values, and that logic should be shared between sync and async, so we didn't feel the need to test that redundantly. The async test is more to check that the request goes in and out correctly. Let me know what you think about that approach.

**Review by rickle-msft — APPROVED** [review]

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/android/storage/blob/BlobTest.java (hunk 1 above)
Comment (vcolin7) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/android/storage/blob/BlobTest.java:303]: Sounds good to me as long as we have good coverage :)

**Review by vcolin7 — APPROVED** [review]

**Pull request merged**